dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -1311,6 +1313,20 @@ protected List<Node> convertCallArguments(
             } else {
                 assert lastParamType instanceof ArrayType
                         : "variable argument formal must be an array";
+                // Handle anonymous constructors that extend a class with an enclosing type.
+                // The logic is borrowed from AnnotatedTypes.adaptParameters(). However,
+                // adaptParameters
+                // does not need this part anymore after mering this PR.
+                if (method.getKind() == ElementKind.CONSTRUCTOR
+                        && method.getEnclosingElement().getSimpleName().contentEquals("")
```

Comment:
```suggestion
                        && ((TypeElement) method.getEnclosingElement()).getNestingKind() == NestingKind.ANONYMOUS
```

Does this work? It's maybe a bit clearer than the string comparison.
We have a [helper method](https://github.com/eisop/checker-framework/blob/b235df2211833a32100d3a78c6100a0315fc1d48/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java#L1715) for trees, but not for `ExecutableElement`s.
Is it worth looking whether this pattern is used elsewhere and replacing it with a new helper?

Reply:
Thanks! Let me try first. I believe it is worth creating a new helper method like "isAnoymous" for a contructor ExecutableElement/